### PR TITLE
Add model generator

### LINKED ISSUE #1: Add a generator for creating the User model
The generator should create:

- [ ] A migration with the required columns. See: https://github.com/excid3/revise_auth/blob/main/test/dummy/db/migrate/20230112011459_create_users.rb
- [ ] A User model with `include ReviseAuth::Model`

`rails revise_auth:install`

### PR DESCRIPTION
**Pull request by gathuku**
Add model generator. 

Generate model defaults to `User`

```
rails g revise_auth:model
```

Specify model name 

```
rails g revise_auth:model account 
```

Include Model attributes to be merged with default attributes

```
rails g revise_auth:model account  name:string admin:boolean
```

Example  migration and model files with `rails g revise_auth:model user name:string admin:boolean`

```
# db/migrate/20230113104212_create_users.rb
class CreateUsers < ActiveRecord::Migration[7.0]
  def change
    create_table :users do |t|
      t.string :email
      t.string :password_digest
      t.string :confirmation_token
      t.datetime :confirmed_at
      t.datetime :confirmation_sent_at
      t.string :unconfirmed_email
      t.string :name
      t.boolean :admin

      t.timestamps
    end
    add_index :users, :email, unique: true
  end
end
```

```
# app/models/user.rb
class User < ApplicationRecord
 include ReviseAuth::Model
end
```

closes #1

### CODE REVIEW

File: lib/generators/revise_auth/templates/migration.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+      t.string :first_name
+      t.string :last_name
```
Comment (excid3) [inline lib/generators/revise_auth/templates/migration.rb:5]: Oh, we should remove these.

I was adding them for testing customizing the params in the `test/dummy` app but we don't need to require them.

Maybe we just comment them out with a note that you can add your own? Or we can do like Devise does and allow you to add your own columns when you run the generator?

Comment (gathuku) [reply lib/generators/revise_auth/templates/migration.rb]: @excid3 how does devise handle this. Can something like this work?

Also thought of an optional argument that defaults to `User` for the model name

```
rails g revise_auth:install Account -a first_name:string last_name:string
```

Comment (excid3) [reply lib/generators/revise_auth/templates/migration.rb]: I believe it works by inheriting from the NamedBase generator. https://github.com/heartcombo/devise/blob/main/lib/generators/devise/devise_generator.rb#L7

The NamedBase is what handles the Class name as the first argument if I remember right.

Their ActiveRecord generator is run through the `hook_for :orm` and tehy have attributes as an argument here:
https://github.com/heartcombo/devise/blob/main/lib/generators/active_record/devise_generator.rb#L9

You can also look at the Rails model generator for examples.

We might also want to rename this to `revise_auth:model` which would be more descriptive.

```
rails g revise_auth:model Account first_name:string last_name:string
```

Comment (excid3) [reply lib/generators/revise_auth/templates/migration.rb]: Actually, Noticed also has an example that supports this same thing:

https://github.com/excid3/noticed/blob/master/lib/generators/noticed/model_generator.rb

File: Gemfile [hunk 2]
```diff
@@ -6,7 +6,7 @@ gemspec
 
 gem "puma"
 gem "sqlite3"
-gem "web-console"
+gem "web-console", group: :development
```
Comment (excid3) [inline Gemfile:9]: In gems, the Gemfile is only for development dependencies, so this is kind of redundant.

File: lib/generators/revise_auth/model_generator.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      def model_attributes
+        [
+          "email:string:index",
+          "password_digest:string",
+          "confirmation_token:string",
+          "confirmed_at:datetime",
+          "confirmation_sent_at:datetime",
+          "unconfirmed_email:string"
+        ].concat(
+          attributes.map { "#{_1.name}:#{_1.type}" }
```
Comment (excid3) [inline lib/generators/revise_auth/model_generator.rb:50]: This is a good idea. I think that we will lose any options and indexes though.

For example `account:belongs_to{polymorphic}` or `username:string:index` would be lost. 

I'm curious how Devise does this, but I bet we can override the `initialize` method and intercept the attributes before they're parsed. https://github.com/rails/rails/blob/main/railties/lib/rails/generators/named_base.rb#L17

**Pull request merged**